src/pages/feed/index.vue

```diff
@@ -341,39 +370,66 @@ export default {
         handleClick: () => this.handleCollectiblesTabChange(item.value),
       }));
     },
+    formattedEvents() {
+      return this.sortAndFilterEvents(this.getFolloweeEvents);
+    },
+    pendingMemoFetchList() {
+      return this.formattedEvents.filter(e => !this.getHasFetchMemo(e.tx_hash));
+    },
     displayedEvents() {
-      const uniqueTxHashes = new Set();
-      return this.getFolloweeEvents
-        .reduce((filteredEvents, event) => {
-          if (!uniqueTxHashes.has(event.tx_hash)) {
-            uniqueTxHashes.add(event.tx_hash);
-            filteredEvents.push(event);
-          }
-          return filteredEvents;
-        }, [])
-        .slice(0, this.eventsToShow);
+      if (this.getAvailableFeedTxList) {
+        return this.formattedEvents.filter(e =>
+          this.getAvailableFeedTxList.includes(e.tx_hash)
+        );
+      }
+      return [];
     },
-
     currentView() {
       return this.$route.query.view;
     },
+    shouldShowLoading() {
+      return this.walletFollowees.length && !this.displayedEvents.length;
+    },
+    shouldShowMore() {
+      return this.pendingMemoFetchList.length;
+    },
+    shouldShowEnd() {
+      return this.displayedEvents.length && !this.pendingMemoFetchList.length;
+    },
+    shouldShowEmpty() {
+      return !this.pendingMemoFetchList.length && !this.displayedEvents.length;
+    },
   },
 
   watch: {
-    getAddress(newAddress) {
-      if (newAddress) {
-        this.fetchUserInfo();
-        this.loadNFTClassesForCurrentTabByAddress(this.getAddress);
-        this.fetchNFTDisplayStateListByAddress(this.getAddress);
-        this.updateTopRankedCreators();
-      }
+    loginAddress: {
+      immediate: true,
+      handler(loginAddress) {
+        if (this.getAddress && loginAddress) {
+          this.fetchUserInfo();
+          this.loadNFTClassesForCurrentTabByAddress(loginAddress);
+          this.fetchNFTDisplayStateListByAddress(loginAddress);
+          this.updateTopRankedCreators();
+          this.hasFetchedFirstBatch = false;
+          this.hasFetchedFolloweeEvents = false;
+        }
+      },
     },
     walletFollowees: {
       immediate: true,
       handler(walletFollowees) {
-        if (walletFollowees.length && !this.hasFetchedFollowees) {
+        if (walletFollowees.length && !this.hasFetchedFolloweeEvents) {
           this.fetchFolloweeWalletEvent();
-          this.hasFetchedFollowees = true;
+          this.hasFetchedFolloweeEvents = true;
```

Comment:
should we await the fetch before setting to `true`?

if not i think should rename to `this.hasStartedFetchingFolloweeEvents`